node/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -495,4 +531,35 @@ deriveSafeCopySimple 0 'base ''AddressInfo
 deriveSafeCopySimple 0 'base ''AccountInfo
 deriveSafeCopySimple 0 'base ''WalletTip
 deriveSafeCopySimple 0 'base ''WalletInfo
-deriveSafeCopySimple 0 'base ''WalletStorage
+
+-- Legacy versions, for migrations
+
+data WalletStorage_v0 = WalletStorage_v0
+    { _v0_wsWalletInfos     :: !(HashMap (CId Wal) WalletInfo)
+    , _v0_wsAccountInfos    :: !(HashMap AccountId AccountInfo)
+    , _v0_wsProfile         :: !CProfile
+    , _v0_wsReadyUpdates    :: [CUpdateInfo]
+    , _v0_wsTxHistory       :: !(HashMap (CId Wal) (HashMap CTxId CTxMeta))
+    , _v0_wsHistoryCache    :: !(HashMap (CId Wal) [TxHistoryEntry])
+    , _v0_wsUtxo            :: !Utxo
+    , _v0_wsUsedAddresses   :: !CustomAddresses
+    , _v0_wsChangeAddresses :: !CustomAddresses
+    }
+
+deriveSafeCopySimple 0 'base ''WalletStorage_v0
+deriveSafeCopySimple 1 'extension ''WalletStorage
+
+instance Migrate WalletStorage where
+    type MigrateFrom WalletStorage = WalletStorage_v0
+    migrate WalletStorage_v0{..} = WalletStorage
+        { _wsWalletInfos     = _v0_wsWalletInfos
+        , _wsAccountInfos    = _v0_wsAccountInfos
+        , _wsProfile         = _v0_wsProfile
+        , _wsReadyUpdates    = _v0_wsReadyUpdates
+        , _wsTxHistory       = _v0_wsTxHistory
+        , _wsHistoryCache    = HM.map txHistoryListToMap _v0_wsHistoryCache
+        , _wsUtxo            = _v0_wsUtxo
+        , _wsBalances        = utxoToAddressCoinMap _v0_wsUtxo
```

Comment:
I don't think that you should alter already existing migration - this wouldn't work, because customers already applied first migration. You should remake this PR so that it's based on current tip of `cardano-sl-1.0-exchanges` and adds second migration on top of first one.